waspc/src/Wasp/Generator/DbGenerator/Common.hs

```diff
@@ -60,3 +66,42 @@ dbSchemaChecksumOnLastGenerateFileInDbRootDir = [relfile|schema.prisma.wasp-gene
 
 dbSchemaChecksumOnLastGenerateFileProjectRootDir :: Path' (Rel ProjectRootDir) (File DbSchemaChecksumOnLastGenerateFile)
 dbSchemaChecksumOnLastGenerateFileProjectRootDir = dbRootDirInProjectRootDir </> dbSchemaChecksumOnLastGenerateFileInDbRootDir
+
+data MigrateArgs = MigrateArgs
+  { _migrationName :: Maybe String,
+    _isCreateOnlyMigration :: Bool
+  }
+  deriving (Show, Eq)
+
+emptyMigrateArgs :: MigrateArgs
+emptyMigrateArgs = MigrateArgs {_migrationName = Nothing, _isCreateOnlyMigration = False}
+
+asArgs :: MigrateArgs -> [String]
+asArgs migrateArgs = do
+  concat . concat $
+    [ [["--create-only"] | _isCreateOnlyMigration migrateArgs],
+      [["--name", name] | Just name <- [_migrationName migrateArgs]]
+    ]
+
+parseMigrateArgs :: Maybe [String] -> MigrateArgs
```

Comment:
I would suggest moving this logic into CLI, where the command is parsed.
I would keep `asArgs` here, and I would keep `MigrateArgs` here, but I would move this logic that translates from CLI args to MigrateArgs.
asArgs could maybe go to Jobs/Db though, not 100% sure -> wherever is closer to Prisma.

Reply:
Sounds good, will do. That should simplify the interface of `Maybe [String]` as well 👍🏻